Problem when I build for production in Angular-Cli

When I try execute ng build --prod or ng build --prod --aot always returns the error below:

`
ERROR in Error encountered resolving symbol values statically. Could not resolve type Document (position 40:41 in the original .ts file), resolving symbol MaskDirective in /home/username/Documents/my-project/node_modules/ng2-mask/build/mask.directive.d.ts

ERROR in ./src/main.ts
Module not found: Error: Can't resolve './$$_gendir/app/app.module.ngfactory' in '/home/username/Documents/my-project/src'
 @ ./src/main.ts 4:0-74
 @ multi ./src/main.ts
`
I created an empty project by importing only ng2-mask and got the same problem.

My environment:
`
@angular/cli: 1.1.1
node: 7.0.0
os: linux x64
@angular/animations: 4.1.3
@angular/common: 4.1.3
@angular/compiler: 4.1.3
@angular/core: 4.1.3
@angular/forms: 4.1.3
@angular/http: 4.1.3
@angular/platform-browser: 4.1.3
@angular/platform-browser-dynamic: 4.1.3
@angular/router: 4.1.3
@angular/cli: 1.1.1
@angular/compiler-cli: 4.1.3
@angular/language-service: 4.1.3
`

--- a/src/app/ng2-mask/mask.directive.ts
+++ b/src/app/ng2-mask/mask.directive.ts
@@ -37,7 +37,7 @@ export class MaskDirective implements OnInit, ControlValueAccessor {
   public constructor(
     private _elementRef: ElementRef,
     private _renderer: Renderer2,
-    @Inject(DOCUMENT) private document: Document
+    @Inject(DOCUMENT) private document: any
   ) {
     this.modelWithSpecialCharacters = true;
     this._clearIfNotMatch = false;

